From 49c7acb905531457dfe46a28635930a63835e731 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 2 Jul 2018 16:22:02 +0300
Subject: [PATCH 39/40] healthd: charger: Draw battery capacity in charging
 animation

Draw a text to show the battery capacity value in charging animation.

Change-Id: I2223daf31721b0bde862fb236b35592842cae07d
---
 healthd/healthd_draw.cpp | 20 ++++++++++++++++++--
 healthd/healthd_draw.h   |  2 ++
 2 files changed, 20 insertions(+), 2 deletions(-)

diff --git a/healthd/healthd_draw.cpp b/healthd/healthd_draw.cpp
index ff93f4bd8..fb9e283c5 100644
--- a/healthd/healthd_draw.cpp
+++ b/healthd/healthd_draw.cpp
@@ -47,14 +47,30 @@ HealthdDraw::HealthdDraw(animation* anim)
 
 HealthdDraw::~HealthdDraw() {}
 
+#define STR_LEN    64
+void HealthdDraw::draw_capacity(const animation* batt_anim) {
+    char cap_str[STR_LEN];
+    int x, y;
+    int str_len_px;
+
+    snprintf(cap_str, (STR_LEN - 1), "%d%%", batt_anim->cur_level);
+    str_len_px = gr_measure(gr_sys_font(), cap_str);
+    x = (gr_fb_width() - str_len_px) / 2;
+    y = (gr_fb_height() + char_height_) / 2;
+    gr_color(0xa4, 0xc6, 0x39, 255); // green
+    draw_text(gr_sys_font(), x, y, cap_str);
+}
+
 void HealthdDraw::redraw_screen(const animation* batt_anim, GRSurface* surf_unknown) {
   clear_screen();
 
   /* try to display *something* */
-  if (batt_anim->cur_level < 0 || batt_anim->num_frames == 0)
+  if (batt_anim->cur_level < 0 || batt_anim->num_frames == 0) {
     draw_unknown(surf_unknown);
-  else
+  } else {
     draw_battery(batt_anim);
+    draw_capacity(batt_anim);
+  }
   gr_flip();
 }
 
diff --git a/healthd/healthd_draw.h b/healthd/healthd_draw.h
index 6a6ba7655..f7dd985db 100644
--- a/healthd/healthd_draw.h
+++ b/healthd/healthd_draw.h
@@ -57,6 +57,8 @@ class HealthdDraw {
   virtual void draw_percent(const animation* anim);
   // Draws charger->surf_unknown or basic text.
   virtual void draw_unknown(GRSurface* surf_unknown);
+  // Draws battery percentage text animation.
+  virtual void draw_capacity(const animation* batt_anim);
 
   // Pixel sizes of characters for default font.
   int char_width_;
-- 
2.11.0

